domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt

```diff
@@ -97,6 +101,13 @@ class SpotlightStateController @Inject constructor(
       }
   }
 
+  private fun spotlightStateRetrieverHelper(spotlightViewState: SpotlightViewState):
+    SpotlightViewState {
+      return if (spotlightViewState == SpotlightViewState.SPOTLIGHT_VIEW_STATE_UNSPECIFIED) {
+        SpotlightViewState.SPOTLIGHT_NOT_SEEN
+      } else spotlightViewState
+    }
```

Comment:
Perhaps you could do an extension here, but more importantly the name could be more clear. Consider:

```kotlin
private fun SpotlightViewState.ensureSeenOrNotSeen(): SpotlightViewState =
  if (this == SPOTLIGHT_VIEW_STATE_UNSPECIFIED) SPOTLIGHT_NOT_SEEN else this
```

Reply:
so elegant!❤